### DISCUSSION TITLE
Where gas fee received from user transaction?

### DISCUSSION THREAD

**Discussion by amandatsalis** [0xPolygon/polygon-edge#921] [General]
Where gas fee received from user transaction?
## Description

Question 1 : 
i am case study about blockchain in university, im trying to create blockchain in localhost, but i not understand where gas fee from user or user create smartcontract is pay gas fee, but where gas fee send ? to null address?

Q2 : 
can i setting for received gas fee from user to my address ? 

Q3 : and how to add validator with costume address not null address. 

Q4 : i see CLI Command :  `server --join /ip4/127.0.0.1/tcp/10001/p2p/16Uiu2HAmJxxH1tScDX2rLGSU9exnuvZKNM9SoK3v315azp68DLPW` how to use this command? 

Have 4 Bootnodes, Included 1 JSON RPC

**Comment by laviniat1996** [comment]
Hi @amandatsalis ,

A1: The only reward mechanism Polygon Edge has, which works for both PoA and PoS, is implemented in the executor.go file (from the State folder). The proposer is going to get the coinbase fee. The rewards are stored to the validator's address. The implementation is:
https://github.com/0xPolygon/polygon-edge/blob/d3d4f8ac4f6b401c01e0e8b6afef3d78aa58d8e5/state/executor.go#L486-L488

A2: You can access the gas fees stored to a certain validator's address via its private key. You can import the account into a wallet, for example, using the private key, and then send the tokens to an address of your choice.

A3: I don't fully understand what you mean by this. Every validator gets a unique public address when you generate the keys.

A4: This flag is not used anymore, we will edit the docs accordingly soon.

**Comment by amandatsalis** [comment]
![image](https://user-images.githubusercontent.com/106639635/202344439-6c21d5f6-c531-4b0e-82de-d0a40cff2b31.png)


how to change this address validator ?

**Reply by laviniat1996** [reply]
@amandatsalis Can you please provide a bit more context for that screenshot?

**Reply by laviniat1996** [reply]
@amandatsalis I think that screenshot is from a block explorer. That validator field (or you can call it a miner) is supposed to be the null address, and you can find the explanation for why it is that way [here](https://wiki.polygon.technology/docs/edge/faq/validators#why-do-the-json-rpc-commands-eth_getblockbynumber-and-eth_getblockbyhash-not-return-the-miners-address).

**Comment by amandatsalis** [comment]
Sorry if I often ask, because I'm doing a case study on blockchain, please help.

For example:
in this screenshot their validator is address which is not null. on my localhost blockchain they show null address 0x0000000000000000000000 now I want to change that null address to my address. E.g : 0x348C310221Bba25FEe7D575Bc5D2c90BbCF88191 and get the reward


![image](https://user-images.githubusercontent.com/106639635/202352416-9fe6ae8c-1279-43e0-9da3-31373c8b1604.png)

**Reply by laviniat1996** [reply]
@amandatsalis The miner field in clique algorithm(used by Edge) has a different meaning. Basically, the miner (validator) field is used to indicate a target address for `new signer proposal `or `delete signer proposal`. It doesn't show the proposer of the block, like you see in that example.It will always be the 0 address. Read more about it [here ](https://eips.ethereum.org/EIPS/eip-225).

**Comment by amandatsalis** [comment]
does that mean I can't change the address validator 0000000000000000000000 to my address like 0x348C310221Bba25FEe7D575Bc5D2c90BbCF88191?

**Comment by laviniat1996** [comment]
That field doesn't display the actual address of the proposer.

**Reply by atherkhalil** [reply]
Hi @laviniat1996 I believe what he wants to ask is that, how can we get the block creator's address? You mentioned in #969 that, we can get it from the "extraData" field in the block header. Please guide on how to decode the extraData field. I'd greatly appreciate any help on this.
